### ISSUE TITLE
undefined method `id' for nil:NilClass

### ISSUE THREAD

**Issue by agatsoh** [Codewars/codewars.com#1934]
#### Describe the bug
I was solving a Kata and I ran Sample tests which ran successfully, then I ran `Attempt` which also ran successfully and then I clicked `Submit Final` which gave me this error
```
Looks like there was a server error! (500)
We have taken note of the situation and will be looking into it.

If you would like to supply us with some details as to how this happened please visit our Github issues list. This will help us get things fixed faster.
```

```
undefined method `id' for nil:NilClass
```

#### To Reproduce

Defined in the description itself.

#### Expected Behavior

After `Submit Final` I am usually taken to the Page where I see highly ranked and rated solutions of other code warriors.

#### Screenshots

https://imgur.com/h48HYVR.png

#### Contexts

Please complete the following information:

- Codewars username (if account related): sohan
- OS: Arcolinux(Arch derivative)
- Browser: Chromium
- Browser Version: Version 76.0.3809.132 (Official Build) Arch Linux (64-bit)

**Comment by kazk** [comment]
Thanks for reporting. Did the page show when you refresh?

**Comment by agatsoh** [comment]
Luckily my solution for that particular Kata was accepted and it now shows inside Solutions

**Comment by agatsoh** [comment]
@kazk Unfortunately I did not refresh the page, I went to my profile so cannot say exactly.

**Comment by FArekkusu** [comment]
This is not specific to submitting solutions. Loading any page can result in this error message.

**Comment by kazk** [comment]
@agatsoh thanks

---

It's not related to submitting solutions, but the solutions page has the highest error rates after the database upgrade (mostly because many other error that used to happen more often was fixed).

> Loading any page can result in this error message.

Same error message doesn't mean much. Codewars is still full of `nil` errors like this one and I'm trying to find the error that's associated with this in our error tracking system.

**Comment by Steffan153** [comment]
This happens to me all the time.
It sounds like it tried to retrieve something from the database but failed and then tried to access `id` on `nil`?

**Comment by kazk** [comment]
Yeah, Codewars often assumes something exists and it fails badly. It happens in multiple places for different reasons. It's really difficult to find where something is defined too because of the way the code is written.

The reason why I asked about refreshing the page is because sometimes the data is actually corrupted (like some referenced document no longer exists) and the page constantly crashes.

The solution page has some serious issues (I'm assuming unrelated to this one) that hurts performance so I'm looking into it. I'll be working on (relatively) small bugs like this one next.

**Comment by Steffan153** [comment]
When it happens, a refresh fixes it tho.
But it can happen on any page, not just solutions.
